stl/inc/vector

```diff
@@ -458,6 +494,139 @@ public:
     using difference_type = typename _Alty_traits::difference_type;
 
 private:
+#ifdef _INSERT_VECTOR_ANNOTATION
+    static _CONSTEXPR20 void _Create_annotation(const vector& _Myvec_) noexcept {
+        // Annotates the shadow memory of the valid range
+        auto& _My_data = _Myvec_._Mypair._Myval2;
+        _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Myend, _My_data._Mylast);
+    }
+
+    static _CONSTEXPR20 void _Clear_annotation(const vector& _Myvec_) noexcept {
+        // Removes the shadow memory annotation of the range
+        auto& _My_data = _Myvec_._Mypair._Myval2;
+        _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast, _My_data._Myend);
+    }
+
+    static _CONSTEXPR20 void _Change_annotation(const vector& _Myvec_, const difference_type _Count) noexcept {
+        // Extends /shrinks the annotated range by _Count
+        auto& _My_data = _Myvec_._Mypair._Myval2;
+        _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast, _My_data._Mylast + _Count);
+    }
+
+    static const void* _Get_aligned_first(const void* _First, const void* _End) noexcept {
+        const char* _CFirst    = reinterpret_cast<const char*>(_First);
+        const char* _CEnd      = reinterpret_cast<const char*>(_End);
+        const size_t _Capacity = static_cast<size_t>(_CEnd - _CFirst);
+
+        if (_Capacity >= 8) {
+            // We are guaranteed to have sufficient space to find an aligned address.
+            return reinterpret_cast<void*>((reinterpret_cast<uintptr_t>(_CFirst) + 7) & ~7);
+        }
+
+        uintptr_t _Alignment_offset = reinterpret_cast<uintptr_t>(_CFirst) & 7;
+        if (_Alignment_offset) {

```

Comment:
We conventionally avoid directly testing integers for non-zero-ness, so this should say `!= 0`. (There's an partial exception, where we tend to test `if (_Flags & _IsKitten)` directly, but otherwise we're consistent about this.)